src/main/scala/cromwell/engine/workflow/WorkflowOptions.scala

```diff
@@ -0,0 +1,89 @@
+  private lazy val EncryptedFields: Seq[String] =
+    WorkflowOptionsConf.getList("encrypted-fields").unwrapped().asScala.toSeq.asInstanceOf[Seq[String]]
+  private lazy val EncryptionKey: String =
+    WorkflowOptionsConf.getString("base64-encryption-key")
+
+  def encryptField(value: JsString): Try[JsObject] = {
+    Aes256Cbc.encrypt(value.value.getBytes("utf-8"), SecretKey(EncryptionKey)) match {
+      case Success(encryptedValue) => Success(JsObject(Map(
+        "iv" -> JsString(encryptedValue.base64Iv),
+        "ciphertext" -> JsString(encryptedValue.base64CipherText)
+      )))
+      case Failure(ex) => Failure(ex)
+    }
+  }
+
+  def decryptField(obj: JsObject): Try[String] = {
+    (obj.fields.get("iv"), obj.fields.get("ciphertext")) match {
+      case (Some(iv: JsString), Some(ciphertext: JsString)) =>
+        Aes256Cbc.decrypt(EncryptedBytes(ciphertext.value, iv.value), SecretKey(WorkflowOptions.EncryptionKey)) match {
+          case Success(decryptedBytes) => Success(new String(decryptedBytes, "utf-8"))
+          case Failure(ex) => Failure(ex)
+        }
+      case _ => Failure(new Throwable(s"JsObject must have 'iv' and 'ciphertext' fields to decrypt: $obj"))
+    }
+  }
+
+  def isEncryptedField(jsValue: JsValue): Boolean = jsValue match {
+    case obj: JsObject if obj.fields.keys.exists(_ == "iv") && obj.fields.keys.exists(_ == "ciphertext") => true
+    case _ => false
+  }
+
+  def fromJsonObject(jsObject: JsObject): Try[WorkflowOptions] = {
+    val encrypted = jsObject.fields map {
+      case (k, v: JsString) if EncryptedFields.contains(k) => k -> encryptField(v)
+      case (k, v: JsString) => k -> Success(v)
+      case (k, v) if isEncryptedField(v) => k -> Success(v)
+      case (k, v) => k -> Failure(new Throwable(s"Unsupported key/value pair in WorkflowOptions: $k -> $v"))
+    }
+
+    encrypted.values collect { case f: Failure[_] => f } match {
+      case s if s.nonEmpty => Failure(new Throwable(s.map(_.exception.getMessage).mkString("\n")))
+      case _ => Success(WorkflowOptions(new JsObject(encrypted map { case (k, v) => k -> v.get })))
+    }
+  }
+
+  def fromJsonString(json: String): Try[WorkflowOptions] = Try(json.parseJson) match {
+    case Success(obj: JsObject) => fromJsonObject(obj)
+    case _ => Failure(new Throwable("Expecting a JSON object"))
+  }
+
+  def fromMap(m: Map[String, String]) = fromJsonObject(JsObject(m map { case (k, v) => k -> JsString(v)}))
+}
+
+case class WorkflowOptions(jsObject: JsObject) {
+  import WorkflowOptions._
+
+  def toMap = jsObject.fields
+
+  def get(key: String): Try[String] = jsObject.fields.get(key) match {
+    case Some(jsStr: JsString) => Success(jsStr.value)
+    case Some(jsObj: JsObject) if isEncryptedField(jsObj) => decryptField(jsObj)
+    case x => Failure(new Throwable(s"Unsupported JsValue as value: $x"))
+  }
+
+  def getOrElse[B >: String](key: String, default: ⇒ B): B = get(key) match {
```

Comment:
Granted we did seriously consider something the other day which had an actual lambda character in it ;)